kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -124,6 +125,15 @@ public TransactionBuilder withTableProperties(Engine engine, Map<String, String>
     return this;
   }
 
+  @Override
+  public TransactionBuilder withUnsetTableProperties(Set<String> propertyKeys) {
+    checkArgument(
+        propertyKeys.stream().noneMatch(key -> key.toLowerCase(Locale.ROOT).startsWith("delta.")),
+        "Unsetting 'delta.' table properties is currently unsupported");
+    this.unsetTablePropertiesKeys = Optional.of(Collections.unmodifiableSet(propertyKeys));
```

Comment:
check `unsetTablePropertiesKeys` is not set.

Reply:
I feel like no reason to prevent calling this more than once? It's weird half of our APIs we seem to do a check like this the other half we just set it (like `withTableProperties`)